indy_node/utils/node_control_utils.py

```diff
@@ -357,18 +357,25 @@ def get_deps_tree(cls, *package, depth=0):
         return ret
 
     @classmethod
-    def get_deps_tree_filtered(cls, *package, filter_list=[], depth=0):
+    def get_deps_tree_filtered(cls, *package, hold_list=[], depth=0, deps_map={}):
         ret = list(set(package))
-        filter_list = [f for f in filter_list if not list(filter(lambda x: f in x, ret))]
-        if depth < MAX_DEPS_DEPTH and filter_list:
+        if depth < MAX_DEPS_DEPTH:
             package_info = cls._get_info_from_package_manager(*ret)
             _, deps = cls._parse_version_deps_from_pkg_mgr_output(package_info)
-            deps_deps = []
-            deps = list(set(deps) - set(ret))
-            deps_deps.append(cls.get_deps_tree_filtered(*deps, filter_list=filter_list, depth=depth + 1))
-
-            ret.append(deps_deps)
-        return ret
+            # Make deps list unique
+            deps = list(set(deps + ret))
+            for d in deps:
+                if "=" in d:
+                    p, v = d.split("=")
+                else:
+                    p = d
+                    v = ''
+                if p in hold_list and \
+                        (p not in deps_map or deps_map[p] == ''):
```

Comment:
Do we have unit tests where we have the same dependency in a tree with a version and without a version?

Reply:
Yes, in test `test_deps_levels` we have pkg_a, which depends from `bb` and `cc` with versions, but package `bb` depends from package `cc` without version.